### PR TITLE
protection for concurrent operations on Indicator components

### PR DESCRIPTION
**Pull request by AbakusW**
# Description

This PR adds protections for concurrent operations on:
* MenuNode attributes
* resources of the Indicator that handle graphic operations, e.g. change of the tray icon, change of the tray label and the launch of desktop notifications

### CODE REVIEW

File: internal/tray-agent/app-indicator/indicator.go [hunk 1]
```diff
@@ -247,12 +264,18 @@ func (i *Indicator) SetMenuTitle(title string) {
...
+	gr.RLock()
+	defer gr.RUnlock()
 	return i.icon
 }
 
 //SetIcon sets the Indicator tray icon. If 'ico' is not a valid argument or ico == IconLiqoNil,
 //SetIcon does nothing.
 func (i *Indicator) SetIcon(ico Icon) {
+	gr := i.graphicResource[resourceIcon]
+	gr.Lock()
```
Comment (palexster) [inline internal/tray-agent/app-indicator/indicator.go:277]: Can we lock just the status write operation instead the whole function body?

Comment (AbakusW) [reply internal/tray-agent/app-indicator/indicator.go:277]: yes you are right!

**Review by mlavacca — APPROVED** [review]

**Review by palexster — APPROVED** [review]

**Pull request merged**